src/htr2hpc/train/apps.py

```diff
@@ -0,0 +1,238 @@
+    xml_path.open("w").write(serialize(segmentation, image_size=part.image.size))
+
+
+def compile_data(segmentations, output_dir):
+    output_file = output_dir / "dataset.arrow"
+    build_binary_dataset(
+        files=segmentations, format_type=None, output_file=str(output_file)
+    )
+    return output_file
+
+
+# TODO: figure out how to run this as a parsl python app,
+# in parallel with downloading training data
+def get_model(api, model_id, training_type, output_dir):
+    model_info = api.model_details(model_id)
+    if model_info.job != training_type:
+        raise ValueError(
+            f"Model {model_id} is a {model_info.job} model, but {training_type} requested"
+        )
+    return api.download_file(model_info.file, output_dir)
+
+
+def get_training_data(api, output_dir, document_id, part_ids=None):
+    # if part ids are not specified, get all parts
+    if part_ids is None:
+        doc_parts = api.document_parts_list(document_id)
+        part_ids = [part.pk for part in doc_parts.results]
+
+    # document details includes line and block types
+    document_details = api.document_details(document_id)
+
+    # get segmentation data for each part of the document that is requested
+    segmentation_data = [
+        get_segmentation_data(api, document_details, part_id, output_dir)
+        for part_id in part_ids
+    ]
+    # if we're generating alto-xml (i.e., segmentation training data),
+    # serialize each of the parts we downloaded
+    [serialize_segmentation(seg, part) for (seg, part) in segmentation_data]
+
+    # NOTE: binary compiled data is only supported train and not segtrain
+    # compiled_data = compile_data(segmentations, output_dir)
+
+
+def segtrain(
+    input_data_dir: pathlib.Path,
+    input_model: pathlib.Path,
+    output_model: pathlib.Path,
+    num_workers: int = 8,
+    # optional param to specify name based on document? include date?
+):
+    segtrain_slurm = Slurm(
+        nodes=1,
+        ntasks=1,
+        cpus_per_task=num_workers,
+        mem_per_cpu="4G",
+        gres=["gpu:1"],
+        job_name="segtrain",
+        output=f"segtrain_{Slurm.JOB_ARRAY_MASTER_ID}.out",
+        time=datetime.timedelta(minutes=20),
+        # time=datetime.timedelta(hours=2),
+    )
+    # do we want to use CUDA Multi-Process Service (MPS) ?
+    # della documentation says to specify with --gpu-mps,
+    # but simple slurm doesn't recognize this as a valid directive.
+    # Work around that by adding it as a command (if indeed we want this)
+    # segtrain_slurm.add_cmd("#SBATCH --gpu-mps")
+
+    # add commands for setup steps
+    segtrain_slurm.add_cmd("module purge")
+    segtrain_slurm.add_cmd("module load anaconda3/2024.6")
+    segtrain_slurm.add_cmd("conda activate htr2hpc")
+    logger.debug(f"sbatch file\n: {segtrain_slurm}")
+    # sbatch returns the job id for the created job
+    segtrain_cmd = (
+        f"ketos segtrain --epochs 200 --resize new -i {input_model}"
```

Comment:
Double-checked [default eScriptorium behavior](https://gitlab.com/scripta/escriptorium/-/blob/develop/app/apps/core/tasks.py?ref_type=heads) regarding `--resize`. The default behavior is apparently to use `both`, not `new`. (The difference being the following: say you have a base segmentation model that recognizes regions with classifications "default" and "marginal", and then you fine-tune on it using training data where regions are classified as "MainZone" and "MarginalZone" and "NumberZone". The keyword `both` will result in a model that outputs classifications of all "default", "marginal", "MainZone", "MarginalZone", and "NumberZone". The keyword `new` will result in a model that *only* outputs "MainZone", "MarginalZone", and "NumberZone".)

For the purpose of mimicking default eScriptorium behavior, we should change this to `--resize both` instead. I personally think it would be a better default for it to be `--resize both` when the user selects a model to refine on and `--resize new` when they do not (and so kraken trains on top of blla.mlmodel), but regardless.